integration/container/health_test.go

```diff
@@ -93,6 +93,32 @@ while true; do sleep 1; done
 	poll.WaitOn(t, pollForHealthStatus(ctxPoll, client, id, "healthy"), poll.WithDelay(100*time.Millisecond))
 }
 
+// TestHealthCheckProcessKilled verifies that health-checks exec get killed on time-out.
+func TestHealthCheckProcessKilled(t *testing.T) {
+	skip.If(t, testEnv.OSType == "windows", "FIXME")
+	defer setupTest(t)()
+	ctx := context.Background()
+	client := testEnv.APIClient()
+
+	cID := container.Run(ctx, t, client, func(c *container.TestContainerConfig) {
+		c.Config.Healthcheck = &containertypes.HealthConfig{
+			Test:     []string{"CMD-SHELL", "sleep 30"},
```

Comment:
unfortunately, healthcheck output when interrupted is not captured to be reported by inspect, and replaced by a gerenic timeout message:
```json
            "Health": {
                "Status": "unhealthy",
                "Log": [
                    {
                        "Start": "2022-06-24T08:59:46.200205085Z",
                        "End": "2022-06-24T08:59:47.203354127Z",
                        "ExitCode": -1,
                        "Output": "Health check exceeded timeout (1s)"
                    },
```
This would require the `exec` logic to change, so that it captures the killed process output even in such circumstances. Would this be fine?

Reply:
If the question is; is it ok to add addition text to `Output`,  good question; I'm not aware of any code parsing the output; AFAIK, it's intended purely as "additional info", but the description is a bit ambiguous; https://github.com/moby/moby/blob/c0069b83bce03d1dd1ac2c04bd309f0641bc8114/api/types/types.go#L400


Do we capture the output if the HealthCheck _is_ successful? (e.g. if a successful health check would print `OK`, is that output returned?)

So we could either;

- include both: `Health check exceeded timeout (1s): <output>`
- Or _only_ include `<output>` (which would be a closer match to the description, but I guess the timeout information is still useful

Perhaps we need to update the description of the field to better clarify what the expectation is;

- for success
- on error (is it the raw error output? other output?)

Reply:
Oh right this could be more difficult than just capturing the output because we won't know when the command is completed (or if it ever will).